spec.md

```diff
@@ -0,0 +1,232 @@
+- **`optional-configFormat`** - while `config.mediaType` represents the unique identifier of an artifact, providing a config object is optional. If a config object is provided. `.json`, `.yaml|yml` or other standard textual formats are preferred. By utilizing standard text formats, registry operators MAY parse the contents to provide additional context to users. If the config object is null, the `optional-configFormat` MUST be empty.
+
+### Example `config.mediaTypes`
+
+|Artifact Type|mediaType|
+|-|-|
+| null config | `application/vnd.oci.null-sample.config.v1`|
+| Config as `.json` | `application/vnd.oci.json-sample.config.v2+json`|
+| Config as `.yaml` | `application/vnd.oci.yaml-sample.config.v3+yaml`|
+
+See [visualizing artifacts](#visualizing-artifacts) for additional `mediaType` examples.
+
+## Defining Supported Layer Types
+
+The content of an artifact is represented through a collection of layers and an optional config object. The layers of an Artifact may be independent files, collections of files, or ordinal layered tarballs persisted as blobs within a registries storage provider.
+
+Defining artifact layers involves:
+
+1. [Defining layer content, understanding layer de-duping](#layer-content-format)
+1. [Optional versioning of layer content](#layer-versioning)
+1. [Defining `layer.mediaTypes`](#defining-layermediatypes)
+
+As an example, [OCI Images][image-layer] are represented through an ordinal collection of tar archives. Each blob represents a layer. Each layer overlays the previous layer. Not all artifacts are as large or complex.
+
+Some artifacts may be represented as a single file, such as config artifact representing a deployment details. While other artifacts may include a config object, and a collection of binaries compressed as yet another layer. By separating the layers, the artifact author can benefit from layer de-duplication and concurrent downloading of each layer using various distribution-spec clients. How an artifact client processes layers, merging them with overlay, or extracting them to different locations or hosts is up to the artifact authors.
+
+### Layer Content Format
+
+The content layer format is up to the artifact author and may utilize standard or custom formats. Authors are encouraged to utilize existing formats which may benefit from existing libraries to parse those formats. Authors may choose to persist individual files with standard or custom formats, such as `.json`, `.yaml`, `.bin`, `.sif`, or store a collection of files as a `.tar` archive. Optionally, the author MAY choose to compress the contents. If compression is used, it MUST be inlcuded in the artifact spec so all clients that interact with the artifact type understand what they must process. The extension of the `layer.mediaType` MUST reflect the format of the blob and its optional compression.
```

Comment:
the compression part does not make total sense to me.. compression may be an extension for any type I would think.

Reply:
Removed MUST to avoid the compression/distribution discussion in this spec.